src/api_server/src/lib.rs

```diff
@@ -30,70 +30,13 @@ use mmds::data_store::Mmds;
-#[derive(PartialEq)]
-pub enum VmmAction {
-    /// Configure the boot source of the microVM using as input the `ConfigureBootSource`. This
-    /// action can only be called before the microVM has booted.
-    ConfigureBootSource(BootSourceConfig),
-    /// Configure the logger using as input the `LoggerConfig`. This action can only be called
-    /// before the microVM has booted.
-    ConfigureLogger(LoggerConfig),
-    /// Get the configuration of the microVM.
-    GetVmConfiguration,
-    /// Flush the metrics. This action can only be called after the logger has been configured.
-    FlushMetrics,
-    /// Add a new block device or update one that already exists using the `BlockDeviceConfig` as
-    /// input. This action can only be called before the microVM has booted.
-    InsertBlockDevice(BlockDeviceConfig),
-    /// Add a new network interface config or update one that already exists using the
-    /// `NetworkInterfaceConfig` as input. This action can only be called before the microVM has
-    /// booted.
-    InsertNetworkDevice(NetworkInterfaceConfig),
-    /// Set the vsock device or update the one that already exists using the
-    /// `VsockDeviceConfig` as input. This action can only be called before the microVM has
-    /// booted.
-    SetVsockDevice(VsockDeviceConfig),
-    /// Update the size of an existing block device specified by an ID. The ID is the first data
-    /// associated with this enum variant. This action can only be called after the microVM is
-    /// started.
-    RescanBlockDevice(String),
-    /// Set the microVM configuration (memory & vcpu) using `VmConfig` as input. This
-    /// action can only be called before the microVM has booted.
-    SetVmConfiguration(VmConfig),
-    /// Launch the microVM. This action can only be called before the microVM has booted.
-    StartMicroVm,
-    /// Send CTRL+ALT+DEL to the microVM, using the i8042 keyboard function. If an AT-keyboard
-    /// driver is listening on the guest end, this can be used to shut down the microVM gracefully.
-    #[cfg(target_arch = "x86_64")]
-    SendCtrlAltDel,
-    /// Update the path of an existing block device. The data associated with this variant
-    /// represents the `drive_id` and the `path_on_host`.
-    UpdateBlockDevicePath(String, String),
-    /// Update a network interface, after microVM start. Currently, the only updatable properties
-    /// are the RX and TX rate limiters.
-    UpdateNetworkInterface(NetworkInterfaceUpdateConfig),
-}
 
-/// The enum represents the response sent by the VMM in case of success. The response is either
-/// empty, when no data needs to be sent, or an internal VMM structure.
-#[derive(Debug)]
-pub enum VmmData {
-    /// No data is sent on the channel.
-    Empty,
-    /// The microVM configuration represented by `VmConfig`.
-    MachineConfiguration(VmConfig),
-}
+/// Shorthand type for a request containing a boxed VmmAction.
+pub type ApiRequest = Box<VmmAction>;
```

Comment:
Yes, data going over a `channel` should be boxed if is not small in size. For efficiency reasons, it's better to send a `Box` (pointer) over a `channel` rather than a full-size large object like `enum VmmAction`.

Reply:
That's a good point. I would still favor loosing the Box since most of the `channel` traffic happens when setting up the VM.

Reply:
Clippy complains :tongue:

Reply:
Well then, can't argue with clippy, can we ? :)